tap_graph/src/v2/rav.rs

```diff
@@ -0,0 +1,131 @@
+// Copyright 2023-, Semiotic AI, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+//! # Receipt Aggregate Voucher v2
+
+use std::cmp;
+
+use alloy::{
+    primitives::{Address, Bytes},
+    sol,
+};
+use serde::{Deserialize, Serialize};
+use tap_eip712_message::Eip712SignedMessage;
+use tap_receipt::{
+    rav::{Aggregate, AggregationError},
+    state::Checked,
+    ReceiptWithState, WithValueAndTimestamp,
+};
+
+use super::{Receipt, SignedReceipt};
+
+/// EIP712 signed message for ReceiptAggregateVoucher
+pub type SignedRav = Eip712SignedMessage<ReceiptAggregateVoucher>;
+
+sol! {
+    /// Holds information needed for promise of payment signed with ECDSA
+    ///
+    /// We use camelCase for field names to match the Ethereum ABI encoding
+    #[derive(Debug, Serialize, Deserialize, Eq, PartialEq)]
+    struct ReceiptAggregateVoucher {
+        /// Unique allocation id this RAV belongs to
+        address allocationId;
+        // The address of the payer the RAV was issued by
+        address payer;
+        // The address of the data service the RAV was issued to
+        address dataService;
+        // The address of the service provider the RAV was issued to
+        address serviceProvider;
+        // The RAV timestamp, indicating the latest TAP Receipt in the RAV
+        uint64 timestampNs;
+        // Total amount owed to the service provider since the beginning of the
+        // payer-service provider relationship, including all debt that is already paid for.
+        uint128 valueAggregate;
+        // Arbitrary metadata to extend functionality if a data service requires it
+        bytes metadata;
+    }
+}
```

Comment:
@gusinacio could you just explain why this needs to be abi encoded, presumably because this is interacting with a contract. Would be great to be able to make the link here. Can also do it in a subsequent PR if you wanna explain here quickly 👍

Reply:
`Definition of encodeType`.
https://eips.ethereum.org/EIPS/eip-712#definition-of-domainseparator
https://docs.rs/alloy/latest/alloy/sol_types/macro.sol.html

sol macro generates the encodeType for EIP712, it doesn't use serde and at the time, there was no way to "rename camelCase" like we do in serde

Reply:
it looks like it still doesn't have that https://github.com/alloy-rs/core/issues/570